src/partest/scala/tools/partest/nest/Runner.scala

```diff
@@ -512,7 +512,8 @@ class Runner(val testFile: File, val suiteRunner: SuiteRunner, val nestUI: NestU
     val perGroup = if (testFile.isDirectory) {
       sources.flatMap(f => readOptionsFile(f changeExtension "flags"))
     } else Nil
-    perTest ++ perGroup
+    val perFile  = toolArgsFor(sources)("scalac")
+    perTest ++ perGroup ++ perFile
```

Comment:
I always worried about the cost of checking all these locations. I'd also added the filter checks, which are not cached. I have no idea if it is significant overhead, especially in the CI environment.